Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -427,18 +421,26 @@ Task AcknowledgeReceivedPublishPacket(MqttApplicationMessageReceivedEventArgs ev
             return CompletedTask.Instance;
         }
 
-        async Task<MqttClientConnectResult> AuthenticateAsync(MqttClientOptions options, CancellationToken cancellationToken)
+        async Task<MqttClientConnectResult> Authenticate(MqttClientOptions options, CancellationToken cancellationToken)
         {
             MqttClientConnectResult result;
 
             try
             {
                 var connectPacket = MqttPacketFactories.Connect.Create(options);
+                await Send(connectPacket, cancellationToken).ConfigureAwait(false);
 
-                var connAckPacket = await SendAndReceiveAsync<MqttConnAckPacket>(connectPacket, cancellationToken).ConfigureAwait(false);
+                var receivedPacket = await Receive(cancellationToken).ConfigureAwait(false);
 
-                var clientConnectResultFactory = new MqttClientConnectResultFactory();
-                result = clientConnectResultFactory.Create(connAckPacket, _adapter.PacketFormatterAdapter.ProtocolVersion);
+                if (receivedPacket is MqttConnAckPacket connAckPacket)
+                {
+                    var clientConnectResultFactory = new MqttClientConnectResultFactory();
+                    result = clientConnectResultFactory.Create(connAckPacket, _adapter.PacketFormatterAdapter.ProtocolVersion);
+                }
+                else
+                {
+                    throw new NotSupportedException("Extended authentication handler is not yet supported");
```

Comment:
Just to clarify, extended auth is not supported in the case where the client receives an `AUTH` packet in between sending a `CONNACK` and receiving a `CONNECT` packet, right? But otherwise, MQTT enhanced auth is supported in the following cases:
- Sending authentication method and authentication data in the `CONNECT` packet and immediately receiving a `CONNACK` packet from the server without any `AUTH` packets in between.
- Initiating extended authentication while already connected and having an `AUTH` exchange of arbitrary length until the authentication succeeds (or a disconnection occurs).

@chkr1011 is my understanding right? Sorry if this is a basic question.